src/builder.ts

```diff
@@ -1188,6 +1190,19 @@ export class GatewayBuilder {
         );
 
         break;
+      case SupportedProtocols.Lido:
+          this._metadata.vault = await lido.infos.getVault(
```

Comment:
So in order to support `validatorIndex` in `GatewayState`:

1. Add `validatorIndex` as an optional field in `DepositParams`
2. Add `validatorIndex` field (metadata section)  in `GatewayParams` (this is a work-around)
3. Update gatewayParams: `this.params.validatorIndex = depositParams.validatorIndex`